html/includes/forms/save-host-dependency.inc.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+/*
+ * LibreNMS
+ *
+ * Copyright (c) 2017 Aldemir Akpinar <https://github.com/aldemira/>
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or (at your
+ * option) any later version.  Please see LICENSE.txt at the top level of
+ * the source code distribution for details.
+ */
+header('Content-type: text/plain');
+
+if (is_admin() === false) {
+    die('ERROR: You need to be admin');
+}
+
+if (!is_numeric($_POST['parent_id'])) {
+    echo 'Wrong Parent host ID!';
+    exit;
+}
+
+// A bit of an effort to reuse this code with dependency editing and the dependency wizard (editing multiple hosts at the same time)
+$device_arr = array();
+foreach ($_POST['device_ids'] as $dev) {
+    if (!is_numeric($dev)) {
+        echo 'Wrong device IDs!';
+        exit;
+    } else if ($dev == $_POST['parent_id']) {
+        echo 'A device cannot depend itself';
+        exit;
+    }
+    $device_arr[] = $dev;
+}
+$clause = implode(',', array_fill(0, count($device_arr), '?'));
+
+// So mysql_string should be something like IN('?','?','?') and device_arr is an array of devices_ids.
+// Mysql query should be like this: UPDATE devices set parent_id = $_POST['parent_id'] WHERE device_id IN ('<int>','<int>','<int>'); 
+if (dbQuery('UPDATE `devices` set parent_id = '.$_POST['parent_id'].' WHERE `device_id` IN('.$clause.')', $device_arr)) {
```

Comment:
How come you can't use `dbUpdate()` here?

Reply:
@laf If I remember correctly, 'IN' keyword didn't work correctly with dbUpdate()

Reply:
@aldemira check: dbGenPlaceholders()